### PR TITLE
Enable some scenarios for DynamicRevocationTests on macOS

### PR DESCRIPTION
**Pull request by vcsjones**
With this there are 123 passing scenarios on macOS. The rest of them just seem to be Apple reporting things a little bit differently, and others seem to be some other quirks with CRLs. I'll see if I can chip away at that more.

image

Contributes to #31249

### CODE REVIEW

**Comment by vcsjones** [comment]
These tests are outerloop. If someone could please start osx libraries outerloop, I would appreciate it.

File: src/libraries/System.Security.Cryptography.X509Certificates/tests/RevocationTests/DynamicRevocationTests.cs [hunk 1]
```diff
@@ -20,6 +19,8 @@ public static class DynamicRevocationTests
         private static readonly Oid s_tlsServerOid = new Oid("1.3.6.1.5.5.7.3.1", null);
 
         private static readonly X509ChainStatusFlags ThisOsRevocationStatusUnknown =
+                PlatformDetection.IsOSX ?
+                X509ChainStatusFlags.RevocationStatusUnknown :
                 X509ChainStatusFlags.RevocationStatusUnknown | X509ChainStatusFlags.OfflineRevocation;
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.X509Certificates/tests/RevocationTests/DynamicRevocationTests.cs:24]: This probably means we need to do some tidying up in post-processing, but I'm OK with the tests at least tracking stability for now and uniformity later.  (Clearly that's where they started :smile:)

**Review by bartonjs — APPROVED** [review]

**Comment by vcsjones** [comment]
@bartonjs off topic, when should I start filing issues for Big Sur? (MacOS 10.16 or 11). I realize it's beta but some things look like intentional changes they are unlikely to fix, others I hope are just bugs...

**Comment by bartonjs** [comment]
> when should I start filing issues for Big Sur? (MacOS 10.16 or 11)

I'd just go ahead with it; unless @wfurt has some other plan.

**Comment by wfurt** [comment]
I think we should start filing bugs for 10.16 when we see any problems. We may not jump on them immediately but at least we would see what it is in from of us. It was loosely on my radar and I was planing to install beta as VM for testing after 5.0 is under control.

**Comment by vcsjones** [comment]
@wfurt okay, I'll file some issues. Things mostly work!

**Comment by wfurt** [comment]
> Things mostly work!

For security or in general? With Catalina there were also problems in IO and other areas.

**Comment by vcsjones** [comment]
I... think the macOS X509Certificates outerloop tests failed with an unaligned memory access and created a core dump? https://helix.dot.net/api/2019-06-17/jobs/1cad052d-cc0c-4ff1-8489-f5a664a74279/workitems/System.Security.Cryptography.X509Certificates.Tests/console

**Comment by bartonjs** [comment]
> I... think the macOS X509Certificates outerloop tests failed with an unaligned memory access and created a core dump? 

Yeah, but so did a lot of things.  I tried a rerun and can't find the rerun results from the older results.  Wanna play the empty commit or merge from upstream and push game?

**Comment by vcsjones** [comment]
> Wanna play the empty commit or merge from upstream and push game?

Might as well make the commit useful.

**Comment by vcsjones** [comment]
I saw some other commits that alluded to fixing outerloop, so I thought we'd give it another go with those changes merged in.

**Comment by bartonjs** [comment]
> so I thought we'd give it another go with those changes merged in

I was waiting to see it work on one of my other PRs before suggesting it, but initiative is always appreciated :smile:.

**Comment by bartonjs** [comment]
Ah, apparently I missed that outerloop OSX /did/ run because it had no failures ::grins innocently::... and it was labelled slightly differently than what I searched for.

7 DynamicRevocation tests ran, all passed.

**Pull request merged**